### PR TITLE
Don't require signing keys when doing event stream recovery.

### LINKED ISSUE #6284: Make keys optional for event stream recovery
Event stream recovery currently fails if signing keys are not available. We should make the event stream recovery workflow work even without signing keys.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #6284

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/RecoveryPlatform.java [hunk 1]
```diff
@@ -68,13 +68,21 @@ public class RecoveryPlatform implements Platform, AutoCloseableNonThrowing {
      * @param initialState  the starting signed state
      * @param selfId        the ID of the node
      */
-    public RecoveryPlatform(final Configuration configuration, final SignedState initialState, final long selfId) {
+    public RecoveryPlatform(
+            final Configuration configuration,
+            final SignedState initialState,
+            final long selfId,
+            final boolean loadSigningKeys) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/RecoveryPlatform.java:75]: missing javadoc for new param

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java [hunk 2]
```diff
@@ -213,7 +215,8 @@ public static SignedState reapplyTransactions(
             final SwirldMain appMain,
             final IOIterator<Round> roundIterator,
             final long finalRound,
-            final long selfId)
+            final long selfId,
+            final boolean loadSigningKeys)
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java:219]: missing javadoc for new param

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/RecoveryPlatform.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/RecoveryPlatform.java:75]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java:219]: added

**Review by alittley — APPROVED** [review]

**Review by artemananiev — APPROVED** [review]

**Pull request merged**